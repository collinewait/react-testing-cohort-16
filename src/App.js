import React from 'react';

const App = ({handleChange, answer, handleSubmit}) => {
  const appStyles = {
    color: 'red'
  }
  return (
    <div className="App">
      <form>
        Addition: <input id="value1" type="number" onChange={handleChange} name="value1" />
         + <input id="value2" onChange={handleChange} name="value2" />
        <button type="submit" onClick={handleSubmit}>ADD</button>
      </form>
      <br />
      <p id="answer" style={appStyles}>Answer: {answer}</p>
    </div>
  );
}

App.defaultProps = {
  answer: ''
}

export default App;
