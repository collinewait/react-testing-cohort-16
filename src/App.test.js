import React from 'react';
import { shallow } from 'enzyme';

import App from './App';

describe('App', () => {
  const mockFun = jest.fn();
  let mountApp;
  beforeEach(() => {
    mountApp = shallow(<App
      handleChange= {mockFun}
      handleSubmit= {mockFun}
      answer=''
    />
    );
  })
  it('should render without crashing', () => {
    expect(mountApp).toMatchSnapshot(); 
  });

  it('should call handleChange unctionf', () => {
    mountApp.find('#value1').simulate('change');
    expect(mockFun).toHaveBeenCalled();
  })
  it('should call handleChange unctionf', () => {
    mountApp.find('button').simulate('click');
    expect(mockFun).toHaveBeenCalled();
  })
});

describe('Answer', () => {
const mountApp = shallow(<App
  handleChange= {jest.fn()}
  handleSubmit= {jest.fn()}
  answer={20}
/>
);

const answerField = mountApp.find('#answer')
expect(answerField.text()).toEqual('Answer: 20')
});