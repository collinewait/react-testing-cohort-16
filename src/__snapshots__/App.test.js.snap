// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render without crashing 1`] = `
<div
  className="App"
>
  <form>
    Addition: 
    <input
      id="value1"
      name="value1"
      onChange={[MockFunction]}
      type="number"
    />
    + 
    <input
      id="value2"
      name="value2"
      onChange={[MockFunction]}
    />
    <button
      onClick={[MockFunction]}
      type="submit"
    >
      ADD
    </button>
  </form>
  <br />
  <p
    id="answer"
    style={
      Object {
        "color": "red",
      }
    }
  >
    Answer: 
  </p>
</div>
`;
